writeToStream.js

```diff
@@ -520,7 +535,7 @@ function writeString (stream, string) {
  * @api private
  */
 function writeNumber (stream, number) {
-  return stream.write(numCache[number])
+  return stream.write(getNumber(number))
 }
```

Comment:
I've added the change that I had in mind, have a look!

Also, can you add some unit tests for this?